src/appleseed-max-impl/builtinmapsupport.cpp

```diff
@@ -62,7 +62,7 @@ void connect_output_selector(
         OSLTexture* osl_texture = static_cast<OSLTexture*>(input_map);
         auto output_names = osl_texture->get_output_names();
 
-        if (--output_index < output_names.size())
+        if (output_index > 0 && --output_index < output_names.size())
```

Comment:
It's still very unusual and somewhat dangerous to modify `output_index` in the `if` statement.

If we _know_ for sure that `output_index` is always >= 1, then what I'd do in that case is assert (`DbgAssert()` in Max), and then decrease it.

    DbgAssert(output_index >= 1);
    output_index -= 1;

    if (output_index < output_names.size())
    {
        ...
    }

Why is `output_index` always >= 1?